src/network/networkd-util.h

```diff
@@ -36,12 +36,15 @@ typedef enum NetworkConfigState {
         NETWORK_CONFIG_STATE_REMOVING    = 1 << 4, /* e.g. address_remove() is called, but no response is received yet */
 } NetworkConfigState;
 
-static inline usec_t sec16_to_usec(uint16_t sec, usec_t timestamp_usec) {
-        return sec == UINT16_MAX ? USEC_INFINITY : usec_add(timestamp_usec, sec * USEC_PER_SEC);
+static inline usec_t sec_to_usec(uint32_t sec, usec_t timestamp_usec) {
```

Comment:
Some of the DHCP code (both v4 and v6) uses this function too. This change won't break any of that?